spring-integration-file/src/main/java/org/springframework/integration/file/tail/FileTailingMessageProducerSupport.java

```diff
@@ -118,6 +140,68 @@ protected void publish(String message) {
 		}
 	}
 
+
+
+	@Override
+	protected void doStart() {
+		super.doStart();
+		if (this.idleEventInterval > 0) {
+			this.idleEventScheduledFuture = getTaskScheduler().scheduleWithFixedDelay(() -> {
+				long now = System.currentTimeMillis();
+				long lastAlertAt = this.lastNoMessageAlert.get();
+				long lastReceive = this.lastProduce;
+				if (now > lastReceive + this.idleEventInterval
+						&& now > lastAlertAt + this.idleEventInterval
+						&& this.lastNoMessageAlert.compareAndSet(lastAlertAt, now)) {
+					publishIdleEvent(now - lastReceive);
+				}
+			}, this.idleEventInterval);
+		}
+	}
+
+	@Override
+	protected void doStop() {
+		super.doStop();
+		if (this.idleEventScheduledFuture != null) {
+			this.idleEventScheduledFuture.cancel(true);
+		}
+	}
+
+	private void publishIdleEvent(long idleTime) {
+
+		if (this.eventPublisher != null && getFile().exists()) {
+			FileTailingIdleEvent event = new FileTailingIdleEvent(this, this.file, idleTime);
+			this.eventPublisher.publishEvent(event);
+		}
+		else {
+			logger.info("No publisher for idle event");
```

Comment:
This logger doesn't reflect reality any more. We come here even when file doesn't exist. So, we should move `exists` condition inside